src/PkgAuthentication.jl

```diff
@@ -198,22 +199,35 @@ function step(state::NeedRefresh)::Union{HasNewToken, NoAuthentication, Failure}
     # errors are recoverable by just getting a new token:
     if response isa Downloads.Response && response.status == 200
         try
-            body = JSON.parse(String(take!(output)))
-            if haskey(body, "token")
-                return HasNewToken(state.server, body["token"])
+            body = Pkg.TOML.parse(String(take!(output)))
+            let msg = "token refresh response"
+                assert_dict_keys(body, "access_token", "id_token"; msg=msg)
```

Comment:
Theoretically, PkgAuthentication should really concern itself only with `access_token` in my opinion -- [OAuth2 says](https://oauth.net/id-tokens-vs-access-tokens/) that ID Tokens are an OpenID Connect thing, and Access Tokens should be used to authenticate the requests. But the tests are implemented with `id_token`, and it might make sense to be permissive here. So this check here requires at least one of them to be present, but it doesn't care which one.